macros/src/gen.rs

```diff
@@ -194,11 +194,39 @@ impl<'a> ServiceGenerator<'a> {
                 quote! { None }
             };
 
+            let input_schema = match &handler.arg {
+                Some(PatType { ty, .. }) => {
+                    quote! {
+                        Some(::restate_sdk::discovery::InputPayload {
+                            content_type: Some(<#ty as ::restate_sdk::serde::WithContentType>::content_type().to_string()),
+                            json_schema: Some(<#ty as ::restate_sdk::serde::WithSchema>::generate_schema()),
+                            required: Some(true),
+                        })
+                    }
+                }
+                None => quote! {
+                    Some(::restate_sdk::discovery::InputPayload {
+                        content_type: Some(<() as ::restate_sdk::serde::WithContentType>::content_type().to_string()),
+                        json_schema: Some(<() as ::restate_sdk::serde::WithSchema>::generate_schema()),
+                        required: Some(false),
+                    })
+                }
+            };
+
+            let output_ok = &handler.output_ok;
+            let output_schema = quote! {
+                Some(::restate_sdk::discovery::OutputPayload {
+                    content_type: Some(<#output_ok as ::restate_sdk::serde::WithContentType>::content_type().to_string()),
+                    json_schema: Some(<#output_ok as ::restate_sdk::serde::WithSchema>::generate_schema()),
+                    set_content_type_if_empty: Some(false),
+                })
+            };
```

Comment:
In empty output case, only `set_content_type_if_empty` should be set, and schema should not be set instead: https://github.com/restatedev/restate/blob/main/service-protocol/endpoint_manifest_schema.json#L131